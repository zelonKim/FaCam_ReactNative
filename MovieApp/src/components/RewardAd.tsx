import React, { useEffect, useRef, useState } from 'react';
import { ActivityIndicator, Platform, StyleSheet, View } from 'react-native';
import {
  RewardedAdEventType,
  RewardedInterstitialAd,
  TestIds,
} from 'react-native-google-mobile-ads';

import Colors from 'open-color';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    position: 'absolute',
    backgroundColor: Colors.black,
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    alignItems: 'center',
    justifyContent: 'center',
  },
});

const adUnitId = __DEV__
  ? TestIds.REWARDED_INTERSTITIAL
  : Platform.OS === 'ios'
  ? 'ca-app-pub-898349852935325/534523453'
  : 'ca-app-pub-298453435564563/7688578576';

  
const RewardAd = () => {
  const rewardedAdRef = useRef(
    RewardedInterstitialAd.createForAdRequest(adUnitId),
  );

  const [loaded, setLoaded] = useState(false);

  useEffect(() => {
    rewardedAdRef.current.addAdEventListener(RewardedAdEventType.LOADED, () => {
      setLoaded(true);
      console.log('loaded');
    });
    rewardedAdRef.current.load();
  }, []);

  useEffect(() => {
    if (loaded) {
      rewardedAdRef.current.show();
    }
  }, [loaded]);

  if (!loaded) {
    return (
      <View style={styles.container}>
        <ActivityIndicator />
      </View>
    );
  }

  return null;
};

export default RewardAd;
