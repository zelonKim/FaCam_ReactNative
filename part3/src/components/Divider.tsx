import React from "react";
import { View } from "react-native";


// export const Divider = (props) => {
//     return(
//         <View style={{
//                 alignSelf: 'stretch',
//                 borderWidth: 0.5,
//                 marginHorizontal: 24,
//                 borderColor: 'gray',
//          }} /> 
//     )
// }


export const Divider: React.FC = (props) => {
    return(
        <View style={{
                alignSelf: 'stretch',
                borderWidth: 0.5,
                marginHorizontal: 24,
                borderColor: 'gray',
         }} /> 
    )
}