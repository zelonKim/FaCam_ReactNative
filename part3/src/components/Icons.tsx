import React from 'react';
import Ionicons  from 'react-native-vector-icons/Ionicons';


// export const Icon = (props) => {
//         return (
//             <Ionicons 
//                 name={props.iconName} 
//                 size={props.iconSize} 
//                 color={props.iconColor} 
//             />
//         )
// }




export type IconName = string;

export const Icon: React.FC<{
    name: IconName
    size: number,
    color: string
}> = (props) => {
    return (
        <Ionicons 
            name={props.name} 
            size={props.size} 
            color={props.color} 
        />
    )
}