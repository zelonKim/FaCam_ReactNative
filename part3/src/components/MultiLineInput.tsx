import React, { useState } from 'react';
import { View, TextInput } from 'react-native';

// export const MultiLineInput = (props)=>{
//     const [focused, setFocused] = useState(false);

//     return (
//         <View 
//             style={{
//                 alignSelf:'stretch', 
//                 paddingHorizontal:12, 
//                 paddingVertical:8, 
//                 borderRadius:4, 
//                 borderWidth:1, 
//                 borderColor:focused ? 'black':'gray'
//             }}>
//             <TextInput 
//                 autoCorrect={false}
//                 autoCapitalize='none'
//                 value={props.value}
//                 multiline
//                 onChangeText={props.onChangeText}
//                 placeholder={props.placeholder}
//                 onSubmitEditing={props.onSubmitEditing}
//                 style={{fontSize:props.fontSize ?? 20, height:props.height ?? 200 }}
//                 onFocus={()=> setFocused(true)} 
//                 onBlur={()=> setFocused(false)}
//             />
//         </View>
//     )
// }




export const MultiLineInput:React.FC<{
    value: string,
    onChangeText: (text:string) => void,
    placeholder: string,
    onSubmitEditing: () => void,
    fontSize?: number,
    height?: number
}> = (props)=>{
    const [focused, setFocused] = useState(false);

    return (
        <View 
            style={{
                alignSelf:'stretch', 
                paddingHorizontal:12, 
                paddingVertical:8, 
                borderRadius:4, 
                borderWidth:1, 
                borderColor:focused ? 'black':'gray'
            }}>
            <TextInput 
                autoCorrect={false}
                autoCapitalize='none'
                value={props.value}
                multiline
                onChangeText={props.onChangeText}
                placeholder={props.placeholder}
                onSubmitEditing={props.onSubmitEditing}
                style={{fontSize:props.fontSize ?? 20, height:props.height ?? 200 }}
                onFocus={()=> setFocused(true)} 
                onBlur={()=> setFocused(false)}
            />
        </View>
    )
}