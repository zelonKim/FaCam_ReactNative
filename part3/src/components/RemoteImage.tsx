import React from 'react';
import { ImageProps, Image as RNImage, StyleProp } from 'react-native'

// export const RemoteImage = (props) => {
//     return(
//         <RNImage 
//             source={{ uri: props.url }}
//             style={[props.style, {
//                 width: props.width,
//                 height: props.height
//             }]}
//             />
//     )
// }


export const RemoteImage: React.FC<{
    url: string,
    width: number,
    height: number,
    style?: StyleProp<ImageProps>
}> = (props) => {
    return(
        <RNImage 
            source={{ uri: props.url }}
            style={[props.style, {
                width: props.width,
                height: props.height
            }]}
            />
    )
}