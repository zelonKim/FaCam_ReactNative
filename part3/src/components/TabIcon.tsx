import { View } from "react-native";
import { Badge } from "./Badge";
import { Icon, IconName } from "./Icons";
import React from "react";



// export const TabIcon = (props) => {
//         if(props.visibleBadge) {
//             return (
//                 <View>
//                     <Badge fontSize={10}>
//                         <Icon 
//                             name={props.iconName} 
//                             size={20} 
//                             color={props.iconColor} />
//                     </Badge>
//                 </View>
//             )
//         }

//         return(
//             <View>
//                 <Icon 
//                     iconName={props.iconName} 
//                     iconSize={20} 
//                     iconColor={props.iconColor} />
//             </View>
//         )
// }




export const TabIcon:React.FC<{
    visibleBadge: boolean,
    iconName: IconName,
    iconColor: string;
}> = (props) => {
    if(props.visibleBadge) {
        return (
            <View>
                <Badge>
                    <Icon 
                        name={props.iconName} 
                        size={20} 
                        color={props.iconColor} />
                </Badge>
            </View>
        )
    }

    return(
        <View>
            <Icon 
                name={props.iconName} 
                size={20} 
                color={props.iconColor} />
        </View>
    )
}