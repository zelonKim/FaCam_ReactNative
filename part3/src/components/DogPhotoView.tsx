import React, { useCallback, useEffect, useState } from "react";
import { TypeDog } from "../data/TypeDog";
import { RemoteImage } from "./RemoteImage";
import { useWindowDimensions } from "react-native";
import { api } from "../utils/AxiosUtils";


export const DogPhotoView: React.FC = () => {
    const [dog, setDog] = useState<TypeDog | null>(null);
    const {width} = useWindowDimensions();

    const loadDogPhoto = useCallback(async() => {
        const apiResult = await api.get<{message: string; status: string;}>('breeds/image/random');
        const result = apiResult.data;
        setDog({photoUrl: result.message});
    }, [])


    useEffect(() => {
        loadDogPhoto()
    }, [loadDogPhoto])


    if(dog === null) {
        return null;
    }

    return (
        <RemoteImage 
            testID="image" 
            url={dog.photoUrl}  
            width={width * 0.7} 
            height={width * 0.7} 
        />
    )
}