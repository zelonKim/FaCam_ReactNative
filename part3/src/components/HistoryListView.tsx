import React, { useCallback, useEffect, useState } from "react";
import { FlatList, useWindowDimensions } from "react-native";
import { useDispatch, useSelector } from "react-redux";
import { TypeUserDispatch, getUserLikedHistory } from "../actions/user";
import { TypeRootReducer } from "../store";
import { TypeDog } from "../data/TypeDog";
import { TypeUser } from "../data/TypeUser";
import database from '@react-native-firebase/database'
import { Button } from "./Button";
import { RemoteImage } from "./RemoteImage";


export const HistoryListView: React.FC<{onPressItem: (index:number) => void}> = (props) => {
    const {width} = useWindowDimensions();

    const dispatch = useDispatch<TypeUserDispatch>();

    const user = useSelector<TypeRootReducer, TypeUser | null>(store => store.user.user)

    const [likedList, setLikedList] = useState<TypeDog[]>([]);

    
    const loadLikedList = useCallback(async() => {
        if(user === null) return;

        const ref = `history/${user?.uid}`;

        const currentHistory = await database()
                                        .ref(ref)
                                        .once('value')
                                        .then(snapshot => snapshot.val())
        
        const dogList = Object.keys(currentHistory).map(key => {
            const item = currentHistory[key];

            return {
                photoUrl: item.url
            } as TypeDog;
        })

        setLikedList(dogList);
    
    }, [user])



    useEffect(() => {
        try {
            loadLikedList();
        } catch (ex) {
            console.log(ex);
        }
    },[dispatch, loadLikedList])


    return (
        <FlatList<TypeDog>
                data={likedList}
                numColumns={2}
                renderItem={({item, index}) => {
                    return (
                        <Button 
                            testID={`Button${index}`}
                            onPress={() => {
                                props.onPressItem(index);
                            }}>
                            <RemoteImage url={item.photoUrl} width={width*0.5} height={width*0.5} />
                        </Button>
                    )
                }}
            />
    )
}

