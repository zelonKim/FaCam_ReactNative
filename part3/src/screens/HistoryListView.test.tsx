import { render, waitFor } from "@testing-library/react-native"
import { HistoryListView } from "../components/HistoryListView"
import configureStore from 'redux-mock-store'
import thunk from "redux-thunk"
import { Provider } from "react-redux"

jest.mock('@react-native-firebase/database', () => {
    return () => ({
        ref: jest.fn().mockImplementation(() => ({
                once: jest.fn().mockReturnValue(
                    Promise.resolve({
                        val: jest.fn().mockReturnValue({
                            TEST: {
                                url: 'test',
                            }
                        })
                    })
                )
            })
        )
    })
})


describe('HistoryListView 렌더링 테스트', () => {
    const initialState = {};
    const mockStore = configureStore([thunk]);

    //@ts-ignore
    let store;

    beforeAll(() => {
        store = mockStore({
            user: {
                user: {
                    uid: 'TEST_ID'
                }
            }
        })
    })

    test('HistoryListView 렌더링 스냅샷 테스트', async() => {
        const onPressItem = jest.fn();

        const component = render(
            //ts-ignore
            <Provider store={store}>
                <HistoryListView onPressItem={onPressItem} />
            </Provider>
        );

        await waitFor(() => component.findByTestId('Button0'), {timeout: 2000})

        expect(component.toJSON()).toMatchSnapshot();
    })
})