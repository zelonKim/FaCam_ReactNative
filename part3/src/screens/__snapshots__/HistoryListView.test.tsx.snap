// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryListView 렌더링 테스트 HistoryListView 렌더링 스냅샷 테스트 1`] = `
<RCTScrollView
  data={
    [
      {
        "photoUrl": "test",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={0.0001}
  stickyHeaderIndices={[]}
  viewabilityConfigCallbackPairs={[]}
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            {
              "bottom": 0,
              "left": 0,
              "right": 0,
              "top": 0,
            }
          }
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": undefined,
              "paddingHorizontal": undefined,
              "paddingVertical": undefined,
            }
          }
          testID="Button0"
        >
          <Image
            source={
              {
                "uri": "test",
              }
            }
            style={
              [
                undefined,
                {
                  "height": 375,
                  "width": 375,
                },
              ]
            }
          />
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
